### PR TITLE
Adding mariadb-connector-c

### PR DESCRIPTION
**Pull request by avgra3**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by avgra3** [comment]
@conda-forge/help-c-cpp, ready for review! Thank you.

**Comment by avgra3** [comment]
@conda-forge-admin, please ping team

**Comment by avgra3** [comment]
@conda-forge/staged-recipes this is ready for review!

**Comment by avgra3** [comment]
@conda-forge-admin, please ping team

File: recipes/mariadb-connector-c/bld.bat [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      -DCMAKE_CXX_FLAGS="-I%LIBRARY_INC%" ^
+      -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ^
+      -DINSTALL_DOCREADMEDIR_STANDALONE="%cd%/junk" ^
+      -DINSTALL_DOCDIR="%cd%/junk" ^
+      ..
+if errorlevel 1 exit 1
+
+ninja
+if errorlevel 1 exit 1
+
```
Comment (ocefpaf) [inline recipes/mariadb-connector-c/bld.bat:17]: Is there a make check or ctest that we can run here?

File: recipes/mariadb-connector-c/build.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+cd build
+
+cmake ${CMAKE_ARGS} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+    -DCMAKE_SKIP_INSTALL_ALL_DEPENDENCY=ON \
+    ..
+
+make -k -j${CPU_COUNT}
+
```
Comment (ocefpaf) [inline recipes/mariadb-connector-c/build.sh:14]: Same question as above: Is there a make check or ctest that we can run here?

File: recipes/mariadb-connector-c/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/mariadb-corporation/mariadb-connector-c/archive/v{{ version }}.tar.gz  # [unix]
+  sha256: c0fda1fa6e52dc85de27156cd847088a72d40d9de6514f7efa57c8d93134a54c  # [unix]
+
+  url: https://github.com/mariadb-corporation/mariadb-connector-c/archive/v{{ version }}.zip  # [win]
+  sha256: b945adcdab6ea7369ceb082166cbb703ba1c98cb7d710fd5b6213c8b709e2e6a  # [win]
```
Comment (ocefpaf) [inline recipes/mariadb-connector-c/meta.yaml:13]: Are there any real differences in the sources? Or was it just b/c of the compression? If it was the compression you don't need this.

```suggestion
  url: https://github.com/mariadb-corporation/mariadb-connector-c/archive/v{{ version }}.tar.gz
  sha256: c0fda1fa6e52dc85de27156cd847088a72d40d9de6514f7efa57c8d93134a54c
```

Comment (avgra3) [reply recipes/mariadb-connector-c/meta.yaml:13]: There are no differences. I used another accepted recipe as an example.

File: recipes/mariadb-connector-c/bld.bat (hunk 1 above)
Comment (avgra3) [reply recipes/mariadb-connector-c/bld.bat:17]: Contained within the build there is a directory called unittest with two sub directories libmariadb and mytap.

I think these would be the tests that you're referring to. I am unsure of how to actually add that to the build script.

Comment (ocefpaf) [reply recipes/mariadb-connector-c/bld.bat:17]: Take a look at https://github.com/mariadb-corporation/mariadb-connector-c/blob/3.3/travis.sh for some inspiration.

File: recipes/mariadb-connector-c/build.sh (hunk 2 above)
Comment (avgra3) [reply recipes/mariadb-connector-c/build.sh]: I added ctest to build.sh and build.bat

File: recipes/mariadb-connector-c/bld.bat (hunk 1 above)
Comment (avgra3) [reply recipes/mariadb-connector-c/bld.bat:18]: I added ctest to build.sh and build.bat

File: recipes/mariadb-connector-c/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/mariadb-corporation/mariadb-connector-c/archive/v{{ version }}.tar.gz
+  sha256: c0fda1fa6e52dc85de27156cd847088a72d40d9de6514f7efa57c8d93134a54c
+
+build:
+  number: 0
+  ignore_run_exports:
+      - libcxx  # [osx]
```
Comment (ocefpaf) [inline recipes/mariadb-connector-c/meta.yaml:15]: I'm not sure about this part. @isuruf do we need this here?

Comment (avgra3) [reply recipes/mariadb-connector-c/meta.yaml:15]: @ocefpaf should I remove this from the meta.yaml file?

Comment (ocefpaf) [reply recipes/mariadb-connector-c/meta.yaml:15]: Let's try and test if you really need it.

Comment (avgra3) [reply recipes/mariadb-connector-c/meta.yaml]: I updated the meta file and it looks like we did not need it for osx

**Pull request merged**